src/theory/quantifiers/instantiate.h

```diff
@@ -146,6 +146,29 @@ class Instantiate : public QuantifiersUtil
                         bool mkRep = false,
                         bool modEq = false,
                         bool doVts = false);
+  /**
+   * Same as above, but we also compute a vector failMask which contains
+   * information about what values in terms led to the instantiation not being
+   * added when this method returns false.  For example, if q is the formula
+   *   forall xy. x>5 => P(x,y)
+   * If terms = { 4, 0 }, then this method will return false since
+   *   4>5 => P(4,0)
+   * is entailed true based on rewriting. This method may additionally set
+   * failMask to "10", indicating that x's value was critical, but y's value
+   * was not. In other words, all instantiations including { x -> 4 } will also
+   * lead to this method returning false.
+   *
```

Comment:
It'd be good to say a bit how the method is trying to set the mask by checking one variable at a time in reverse order. And that this matter because of how the mask is used by the user of this method.